langchain4j-embeddings-bge-small-zh-v15/src/test/java/dev/langchain4j/model/embedding/bge/small/zh/v15/BgeSmallZhV15EmbeddingModelTest.java

```diff
@@ -0,0 +1,85 @@
+package dev.langchain4j.model.embedding.bge.small.zh.v15;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.CosineSimilarity;
+import dev.langchain4j.store.embedding.RelevanceScore;
+import org.junit.jupiter.api.Test;
+
+import static dev.langchain4j.internal.Utils.repeat;
+import static dev.langchain4j.model.embedding.internal.VectorUtils.magnitudeOf;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.Percentage.withPercentage;
+
+class BgeSmallZhV15EmbeddingModelTest {
+
+    @Test
+    void should_embed() {
+
+        EmbeddingModel model = new BgeSmallZhV15EmbeddingModel();
+
+        Embedding first = model.embed("你好").content();
+        assertThat(first.vector()).hasSize(512);
+
+        Embedding second = model.embed("您好").content();
+        assertThat(second.vector()).hasSize(512);
+
+        double cosineSimilarity = CosineSimilarity.between(first, second);
+        assertThat(RelevanceScore.fromCosineSimilarity(cosineSimilarity)).isGreaterThan(0.95);
+    }
+
+    @Test
+    void embedding_should_have_the_same_values_as_embedding_produced_by_sentence_transformers_python_lib() {
+
+        EmbeddingModel model = new BgeSmallZhV15EmbeddingModel();
+
+        Embedding embedding = model.embed("书").content();
+
+        assertThat(embedding.vector()[0]).isCloseTo(-0.0003102528f, withPercentage(1));
```

Comment:
How did you get these numbers?

Reply:
Just wrote the code according to the [documentation](https://github.com/FlagOpen/FlagEmbedding/tree/master/FlagEmbedding/llm_embedder) and tested it.

Here is my test code:
```python
import torch
from transformers import AutoTokenizer, AutoModel


def test_bge_small_zh_v15():
    queries = ['书']
    model_name = 'bge-small-zh-v15'

    # from repo on huggingface.co
    # tokenizer = AutoTokenizer.from_pretrained(f'BAAI/{model_name}')
    # model = AutoModel.from_pretrained(f'BAAI/{model_name}')

    # from local filesystem
    tokenizer = AutoTokenizer.from_pretrained(f'./{model_name}/')
    model = AutoModel.from_pretrained(f'./{model_name}/')

    query_inputs = tokenizer(queries, padding=False, return_tensors='pt')
    query_outputs = model(**query_inputs)
    query_embeddings = query_outputs.last_hidden_state[:, 0]

    with torch.no_grad():
        # Normalize
        query_embeddings = torch.nn.functional.normalize(query_embeddings, p=2, dim=1)

    print(
        f'[{queries[0]}]: {query_embeddings[0][0]} {query_embeddings[0][1]} {query_embeddings[0][510]} {query_embeddings[0][511]}')
``` 

Then, the output is
```shell
[书]: -0.00031016150023788214 0.06835819780826569 0.07749755680561066 -0.04785729572176933
```